impl/src/expand.rs

```diff
@@ -203,13 +204,14 @@ fn impl_enum(input: Enum) -> TokenStream {
                 {
                     let backtrace = &backtrace_field.member;
                     let source = &source_field.member;
+                    let source_literal = quote! { source };
                     let source_backtrace = if type_is_option(source_field.ty) {
                         quote_spanned! {source.span()=>
-                            source.as_ref().and_then(|source| source.as_dyn_error().backtrace())
+                            #source_literal.as_ref().and_then(|source| #source_literal.as_dyn_error().backtrace())
```

Comment:
This looks wrong -- I am going to turn this back into `|source| source.as_dyn_error().backtrace()` in the merge.

Reply:
Good catch! Sorry about that.